core/transaction_pool.go

```diff
@@ -180,30 +188,29 @@ func (pool *TxPool) SetLocal(tx *types.Transaction) {
 // validateTx checks whether a transaction is valid according
 // to the consensus rules.
 func (pool *TxPool) validateTx(tx *types.Transaction) error {
-	// Validate sender
-	var (
-		from common.Address
-		err  error
-	)
-
 	local := pool.localTx.contains(tx.Hash())
 	// Drop transactions under our own minimal accepted gas price
 	if !local && pool.minGasPrice.Cmp(tx.GasPrice()) > 0 {
 		return ErrCheap
 	}
 
-	// Validate the transaction sender and it's sig. Throw
-	// if the from fields is invalid.
-	if from, err = tx.From(); err != nil {
+	currentState, err := pool.currentState()
+	if err != nil {
+		return err
+	}
+
+	var from common.Address
+	if pool.homestead {
+		from, err = tx.From()
+	} else {
+		from, err = tx.FromFrontier()
+	}
+	if err != nil {
```

Comment:
OK. There's no need to sample then, since all major clients (Go, CPP, Python, JS) already only create signatures with canonical s values (in lower half of range allowed in ECDSA). It's OK to ignore such txs in the tx pool, even before the homestead release, because if someone is using a client that produces s values in the other half, they can simply retry signing until they get a s value in the lower half. Or better, they can change to a client recommended by the community.

The critical thing is block processing, not tx pool policy.